src/jit/hwintrinsiccodegenxarch.cpp

```diff
@@ -716,11 +716,153 @@ void CodeGen::genSSE2Intrinsic(GenTreeHWIntrinsic* node)
             assert(op1 != nullptr);
             assert(op2 != nullptr);
             assert(baseType == TYP_DOUBLE);
+            ins = Compiler::insOfHWIntrinsic(intrinsicID, baseType);
+            op2Reg          = op2->gtRegNum;
+            ival            = Compiler::ivalOfHWIntrinsic(intrinsicID);
+            emit->emitIns_SIMD_R_R_R_I(ins, emitTypeSize(TYP_SIMD16), targetReg, op1Reg, op2Reg, ival);
+
+            break;
+        }
+
+        case NI_SSE2_CompareEqualOrderedScalar:
```

Comment:
These should be merged with the SSE implementations.

Reply:
Agree, but since there quite a lot refactoring opportunities still coming I have created issue to track this work item #16014 and I plan to work once all SSE2 and other similar SSE intrinsics are in.